user-interface/src/lib/components/Pill.tsx

```diff
@@ -0,0 +1,53 @@
+import './Pill.scss';
+import Icon from './uswds/Icon';
+import { useEffect, useState } from 'react';
+
+const defaultColor = 'black';
+const defaultBackgroundColor = '#d0d0d0';
+
+type PillProps = {
+  id?: string;
+  className?: string;
+  color?: string;
+  backgroundColor?: string;
+  label: string;
+  ariaLabelPrefix?: string;
+  value: string;
+  onClick: (value: string) => void;
+  disabled?: boolean;
+};
+
+export function Pill(props: PillProps) {
+  const [color, setColor] = useState<string>(defaultColor);
+  const [backgroundColor, setBackgroundColor] = useState<string>(defaultBackgroundColor);
+
+  function handleKeyDown(ev: React.KeyboardEvent) {
+    if (ev.key === 'Enter' || ev.key === ' ') {
+      props.onClick(props.value);
+      ev.preventDefault();
+    }
+  }
+
+  useEffect(() => {
+    setColor(props.color ?? defaultColor);
+    setBackgroundColor(props.backgroundColor ?? defaultBackgroundColor);
+  }, []);
+
+  return (
+    <button
+      id={props.id ?? ''}
+      data-testid={props.id ? `pill-${props.id}` : ''}
+      className="pill usa-button--unstyled"
+      style={{ backgroundColor, color }}
```

Comment:
I would just set the colors from the props/defaults and NOT use state variables which will cause refresh once the useEffect is called.

Reply:
Great point.